### PR TITLE
docs: Update code blocks across website

### PR DESCRIPTION
**Pull request by blake**
* Use CodeTabs for examples in multiple formats.
* Ensure correct language on code fences.
* Use CodeBlockConfig for examples with filenames, or which need highlighted content.

A similar changed was recently merged to Learn in hashicorp/learn#3739.

Preview site is available at <https://consul-oa2h8hxsv-hashicorp.vercel.app/>.

### CODE REVIEW

**Review by lkysow — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/425770.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 99b1d8ed8ce771e7db9abe6cf90d447f7431d59f onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 99b1d8ed8ce771e7db9abe6cf90d447f7431d59f onto `release/1.10.x` succeeded!

File: website/content/docs/nia/terraform-modules.mdx [hunk 1]
```diff
@@ -36,6 +36,8 @@ First, create a directory to organize Terraform configuration files that make up
 
 The `main.tf` is the entry point of the module and this is where you can begin authoring your module. It can contain multiple Terraform resources related to an automation task that uses Consul service discovery information, particularly the required [`services` input variable](#services-variable). The code example below shows a resource using the `services` variable. When this example is used in automation with Consul-Terraform-Sync, the content of the local file would dynamically update as Consul service discovery information changes.
 
+<CodeBlockConfig filename="main.tf">
```
Comment (findkim) [inline website/content/docs/nia/terraform-modules.mdx:39]: TIL! this is a really nice QOL addition. Thanks @blake 
image

**Pull request merged**